File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] arr, int target) {
        HashMap<Integer, Integer> hashMap = new HashMap<>();
        for(int i = 0 ; i < arr.length ; i++){
            hashMap.put(arr[i],i);
        }
        System.out.println(hashMap);
        
        for(int i = 0 ; i < arr.length ; i++){
            int element = target - arr[i];
            if(hashMap.containsKey(element) &&hashMap.get(element) != i ){
                return new int[]{i,hashMap.get(element)};
            }
        }
        
        
        return new int[]{-1,-1};
}
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const hashmap = {};
  for (let i = 0; i < nums.length; i++) {
    const currentmapvalue = hashmap[nums[i]];
    if (currentmapvalue >= 0) {
      return [currentmapvalue, i];
    }
    const numbertofind = target - nums[i];
    hashmap[numbertofind] = i;
  }
  return [];
};